common/src/main/java/org/keycloak/common/crypto/CertificateUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.common.crypto;
+
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+/**
+ * The Class CertificateUtils provides utility functions for generation of V1 and V3 {@link java.security.cert.X509Certificate}
+ *
+ */
+public interface CertificateUtils {
```

Comment:
Is it possible to slightly rename this class, so we don't have 2 classes named `CertificateUtils` ? For example this one can be renamed to `CertificateUtilsProvider` ? Same applies for implementation classes (EG. `CertificateUtilsBC` can be renamed to `BCCertificateUtilsProvider` and the fips can be `BCFipsCertificateUtilsProvider`). WDYT?

Reply:
Will do